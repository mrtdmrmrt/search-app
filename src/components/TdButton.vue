<template>
  <button class="td__button" @click="$emit('click')" :disabled="disabled">
    <slot />
  </button>
</template>

<script>
export default {
  name: "TdButton",
  /**
   * button disabled attribute
   */
  props: {
    disabled: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="scss">
.td__button {
  cursor: pointer;
  background-color: #204080;
  border-radius: 12px;
  font-weight: 700;
  font-size: 18px;
  line-height: 21px;
  color: #ffffff;
  padding: 8px 20px;
  outline: none;
  border: none;
  -webkit-transition: background-color 0.2s ease-in-out;
  -ms-transition: background-color 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out;
  &:hover {
    background-color: #4f75c2;
  }
  &:disabled {
    background-color: #e5e5e5;
    cursor: not-allowed;
  }
}
</style>
