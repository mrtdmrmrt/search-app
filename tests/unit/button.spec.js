import { shallowMount } from "@vue/test-utils";
import TdButton from "@/components/TdButton.vue";

describe("TdButton.vue", () => {
  it("renders props.disabled when passed", () => {
    const disabled = true;
    const wrapper = shallowMount(TdButton, {
      propsData: { disabled },
    });
    expect(wrapper.props().disabled).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("renders slots when passed", () => {
    const wrapper = shallowMount(TdButton, {
      slots: {
        default: "<span>label</span>",
      },
    });
    expect(wrapper.text()).toBe("label");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
