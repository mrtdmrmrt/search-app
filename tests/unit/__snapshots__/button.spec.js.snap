// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TdButton.vue renders props.disabled when passed 1`] = `<button disabled="disabled" class="td__button"></button>`;

exports[`TdButton.vue renders slots when passed 1`] = `<button class="td__button"><span>label</span></button>`;
